src/arctica-greeter.vala

```diff
@@ -1071,12 +1071,24 @@ public class ArcticaGreeter : Object
         /* Set GTK+ settings */
         debug ("Setting GTK+ settings");
         var settings = Gtk.Settings.get_default ();
-        var value = AGSettings.get_string (AGSettings.KEY_THEME_NAME);
+        var agsettings_ = new AGSettings ();
```

Comment:
I don't want the greeter to start in some Gtk default theme or the normal theme and then switch over to high-contrast theming immediately after coming up, because AGSettings say so.

In my tests with this branch, I observed AG coming up in normal theme mode and then blend over into high-contrast mode. This causes to much visual noise for visual impaired users, imho.

Also, setting Adwaita first (gtk default theme) and then the normal theme does not make sense here, either. As we obviously have access to all those static values in AGSettings, I don't see any problem using them before AGSettings is fully initialized.

What we can avoid is the pre-mature initialization of "agsettings_". We could also do a static call like`AGSettings.get_boolean(AGSettings.KEY_HIGH_CONTRAST)`.

Basically, the only change needed here is dropping "agsettings_ initialization and usage. IMHO.

Reply:
Okay, that's fair. I concur.

In that case, though, I'd like to move the agsettings object creation to the place where you use it first, including the comment I've written for it.

Additionally, we should drop all the private `default_*theme` variables, it doesn't make sense to keep them. Querying the default themes won't help us. They are hardcoded in GTK, loaded up by default and loading a different theme will just silently do nothing if it's not available. We can't really detect that. We also don't care, because if the strings are invalid, nothing will happen and GTK will stay on its default.

Reply:
ok, go ahead then with this... Thanks!